src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/matrix_nms/matrix_nms_kernel_ref.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "matrix_nms_kernel_ref.h"
+
+#include <vector>
+
+#include "kernel_selector_utils.h"
+
+namespace kernel_selector {
+
+ParamsKey MatrixNmsKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableInputDataType(Datatype::F32);
+    k.EnableInputDataType(Datatype::INT32);
+
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+
+    k.EnableDifferentTypes();
+    k.EnableAllInputLayout();
+    k.EnableAllOutputLayout();
+    k.EnableTensorOffset();
+    k.EnableTensorPitches();
+    k.EnableBatching();
+    return k;
+}
+
+namespace {
+
+MatrixNmsKernelRef::DispatchData SetDefault(const matrix_nms_params& params, size_t idx) {
+    MatrixNmsKernelRef::DispatchData dispatch_data;
+
+    const auto& input_scores = params.inputs[1];
+    if (idx == 0) {
+        dispatch_data.gws = {input_scores.Batch().v, input_scores.Feature().v, 1};
+        dispatch_data.lws = GetOptimalLocalWorkGroupSizes(dispatch_data.gws, params.engineInfo);
+    } else if (idx == 1) {
+        dispatch_data.gws = {input_scores.Batch().v, 1, 1};
+        dispatch_data.lws = GetOptimalLocalWorkGroupSizes(dispatch_data.gws, params.engineInfo);
+    } else {
+        dispatch_data.gws = {1, 1, 1};
+        dispatch_data.lws = {1, 1, 1};
+    }
+
+    return dispatch_data;
+}
+}  // anonymous namespace
+
+KernelsData MatrixNmsKernelRef::GetKernelsData(const Params& params, const optional_params& options) const {
+    if (!Validate(params, options)) {
+        return {};
+    }
+
+    const size_t kernels_num{3};
+    KernelData kernel_data = KernelData::Default<matrix_nms_params>(params, kernels_num);
+    const matrix_nms_params& new_params = dynamic_cast<const matrix_nms_params&>(*kernel_data.params.get());
+
+    const size_t BOX_INFO_SIZE{16};
+
+    const int batches_num = new_params.inputs[1].Batch().v;
+    const int classes_num = new_params.inputs[1].Feature().v;
+    const int boxes_num = new_params.inputs[0].Feature().v;
+
+    int max_boxes_per_class{boxes_num};
+    if (new_params.nms_top_k >= 0)
```

Comment:
Nit: Maybe better to use parentheses even for one line conditions?

Reply:
It's question of many holy-wars) let's postpone this until we have code style)